packages/shared/components/atoms/tiles/ActivityTile.svelte

```diff
@@ -24,6 +25,14 @@
         activity?.direction === ActivityDirection.In && asyncStatus === ActivityAsyncStatus.Unclaimed
     $: timeDiff = activity?.getTimeDiffUntilExpirationTime($time)
 
+    async function handleClaim(activity: Activity): Promise<void> {
+        try {
+            await checkStronghold(() => claimActivity(activity))
```

Comment:
we need to do this also in the `ActivityDetailsPopup`. Maybe its better if we move `checkStronghold` into the `claimActivity` because we never claim activity without checking stronghold